oe_authentication.module

```diff
@@ -10,10 +10,19 @@ declare(strict_types=1);
 /**
  * Implements hook_user_cancel_methods_alter().
  *
- * Remove the option to delete users while keeping the option to block them.
+ * Conditionally removes the option to delete users.
  */
 function oe_authentication_user_cancel_methods_alter(&$methods) {
-  if (\Drupal::currentUser()->id() == 1) {
+  $current_user = \Drupal::currentUser();
+  $restrict_user_delete = \Drupal::configFactory()
+    ->get('oe_authentication.settings')
+    ->get('restrict_user_delete_cancel_methods');
+  // @todo The restriction should not check for UID 1 as it's bad practice. It
+  // should either check for the user being an admin, or use a permission based
+  // approach like the module drupal/user_cancel_methods_permissions.
+  // Furthermore, the 'access' property of the cancel methods should be used,
+  // rather than removing the method altogether.
+  if ($current_user->id() == 1 || $restrict_user_delete !== TRUE) {
```

Comment:
Not sure if we need `!== TRUE`, if the config value is boolean.
It's a matter of trust, convention and expressiveness.
Trust that we always get boolean thanks to the config schema.
Expressiveness in showing that yes we expect a boolean TRUE here.

I say keep it (because I don't have strong opinion otherwise).